editoast/src/views/work_schedules.rs

```diff
@@ -13,25 +13,33 @@ use std::result::Result as StdResult;
 use thiserror::Error;
 use utoipa::ToSchema;
 
+use crate::core::pathfinding::TrackRange as CoreTrackRange;
 use crate::error::InternalError;
 use crate::error::Result;
 use crate::models::prelude::*;
 use crate::models::work_schedules::WorkSchedule;
 use crate::models::work_schedules::WorkScheduleGroup;
 use crate::models::work_schedules::WorkScheduleType;
+use crate::views::path::projection::PathProjection;
 use crate::views::AuthorizationError;
 use crate::views::AuthorizerExt;
 use crate::AppState;
-use editoast_schemas::infra::TrackRange;
+use editoast_schemas::infra::{Direction, TrackRange};
 
 crate::routes! {
-    "/work_schedules" => create,
+    "/work_schedules" => {
+        create,
+        "/project" => project,
+    },
 }
 
 editoast_common::schemas! {
     WorkScheduleCreateForm,
     WorkScheduleCreateResponse,
+    WorkScheduleProjectForm,
+    WorkScheduleProjection,
     WorkScheduleItemForm,
+    WorkScheduleType,
```

Comment:
```suggestion
```

You can `inline` this one as it's a small enum with a single ref. (Actually you can inline all the schemas you created since they're only used once.)

Reply:
done